### PR TITLE
✨ [RUMF-1074] bridge host checking

### PR DESCRIPTION
**Pull request by amortemousque**
## Motivation

The mobile SDKs (Android and iOS) want to track events that happen inside a webview using the browser SDK. To do so the browser SDK will send events to a bridge injected by the mobile SDK. 

Because WebView can be used to display websites not owned by the customer, we need to guarantee that if these websites also have the browser SDK, they won’t send their events to the mobile SDK. 

## Changes

Add webView host checking

## Testing

- [x] Local
- [x] Staging
- [x] Unit
- [x] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/core/src/transport/eventBridge.ts [hunk 1]
```diff
@@ -1,25 +1,35 @@
...
+      eventBridgeGlobal.send(JSON.stringify({ eventType, event }))
     },
   }
 }
 
 export function isEventBridgePresent(): boolean {
-  return !!getEventBridgeGlobal()
+  return getEventBridge()
+    .getAllowedWebViewHosts()
+    .some((o) => includes(o, window.location.hostname))
```
Comment (bcaudan) [inline packages/core/src/transport/eventBridge.ts:32]:
```suggestion
    .some((host) => includes(host, window.location.hostname))
```

Comment (bcaudan) [reply packages/core/src/transport/eventBridge.ts:32]: why `includes` and not `===`? 
should we add a test for it?

Comment (amortemousque) [reply packages/core/src/transport/eventBridge.ts]: It was to handle port but know that I see it, we might have some issue with the include. I've double checked with Marius and an equality will do.

Comment (amortemousque) [reply packages/core/src/transport/eventBridge.ts]: For the test it's done within the `isEventBridgePresent`.

File: packages/core/src/transport/eventBridge.ts [hunk 2]
```diff
@@ -5,21 +5,31 @@ export interface BrowserWindow extends Window {
...
+      eventBridgeGlobal.send(JSON.stringify({ eventType, event }))
     },
   }
 }
 
 export function isEventBridgePresent(): boolean {
-  return !!getEventBridgeGlobal()
+  return getEventBridge()
+    .getAllowedWebViewHosts()
+    .some((host) => host === window.location.hostname)
```
Comment (BenoitZugmeyer) [inline packages/core/src/transport/eventBridge.ts:32]:
```suggestion
  return includes(window.location.hostname, getEventBridge().getAllowedWebViewHosts())
```

File: packages/core/src/transport/eventBridge.ts [hunk 3]
```diff
@@ -5,21 +5,31 @@ export interface BrowserWindow extends Window {
...
   send(msg: string): void
 }
 
 export function getEventBridge<T, E>() {
-  const datadogEventBridge = getEventBridgeGlobal()
+  const eventBridgeGlobal = getEventBridgeGlobal()
+
+  if (!eventBridgeGlobal) {
+    return { getAllowedWebViewHosts: () => [], send: () => undefined }
+  }
```
Comment (BenoitZugmeyer) [inline packages/core/src/transport/eventBridge.ts:17]: Do we need to return a fake event bridge here? Why not returning `undefined`?

File: packages/core/src/transport/eventBridge.spec.ts [hunk 4]
```diff
@@ -3,6 +3,8 @@ import { deleteEventBridgeStub, initEventBridgeStub } from '../../test/specHelpe
 import { getEventBridge, isEventBridgePresent } from './eventBridge'
 
 describe('isEventBridgePresent', () => {
+  const notAllowedWebViewHosts = ['foo.bar']
```
Comment (BenoitZugmeyer) [inline packages/core/src/transport/eventBridge.spec.ts:6]: Shouldn't it be
```suggestion
  const allowedWebViewHosts = ['foo.bar']
```
?

File: packages/core/src/transport/eventBridge.ts (hunk 3 above)
Comment (amortemousque) [reply packages/core/src/transport/eventBridge.ts:17]: Sure it can work.

File: packages/core/src/transport/eventBridge.ts (hunk 2 above)
Comment (amortemousque) [reply packages/core/src/transport/eventBridge.ts:32]: nice!

File: packages/core/src/transport/eventBridge.ts [hunk 5]
```diff
@@ -1,25 +1,34 @@
...
-      datadogEventBridge?.send(JSON.stringify({ eventType, event }))
+      eventBridgeGlobal.send(JSON.stringify({ eventType, event }))
     },
   }
 }
 
 export function isEventBridgePresent(): boolean {
-  return !!getEventBridgeGlobal()
+  const bridge = getEventBridge()
+  return !!bridge && includes(bridge.getAllowedWebViewHosts(), window.location.hostname)
```
Comment (bcaudan) [inline packages/core/src/transport/eventBridge.ts:31]: we don't check only if the bridge is present now.
what about renaming the function to something like `canUseEventBridge`?

**Review by bcaudan — APPROVED** [review]

File: packages/core/src/transport/eventBridge.ts (hunk 5 above)
Comment (amortemousque) [reply packages/core/src/transport/eventBridge.ts:31]: yes sure

**Pull request merged**